internal/setup/setup.go

```diff
@@ -173,20 +207,41 @@ func (r *Release) validate() error {
 		return err
 	}
 
-	// Check for glob conflicts.
-	for newPath, new := range globs {
-		for oldPath, old := range paths {
-			if new.Package == old.Package {
+	checkConflict := func(old, new pathSlice) error {
+		if strdist.GlobPath(new.path, old.path) {
+			if (old.slice.Package > new.slice.Package) || (old.slice.Package == new.slice.Package && old.slice.Name > new.slice.Name) {
+				old, new = new, old
+			}
+			return fmt.Errorf("slices %s and %s conflict on %s and %s", old.slice, new.slice, old.path, new.path)
+		}
+		return nil
+	}
+	for _, new := range globs {
+		// TODO replace with slices.Concat once we upgrade to go 1.22.
+		for _, old := range append(globs, copies...) {
+			if new.slice.Package == old.slice.Package {
```

Comment:
Let's probably add a comment here saying that we do not care for conflicts between globs and copy-paths within the same package.

Reply:
See https://github.com/canonical/chisel/pull/143#discussion_r1666987367, let's keep the discussion about adding comments like that there.